src/pages/RecipeDetail.js

```diff
@@ -1,26 +1,28 @@
+import {BiCategoryAlt} from 'react-icons/bi';
+import {BsClockHistory} from 'react-icons/bs';
+import {FiAlignJustify} from 'react-icons/fi';
 import {useParams} from 'react-router-dom';
 import styled from 'styled-components';
 
-import {getFromLocal} from '../lib/localStorage';
-
-export default function RecipeDetail() {
-  let params = useParams();
-  const local = getFromLocal('recipes');
-
-  const recipeData = local.filter(singleRecipe => singleRecipe.id === params.id);
-
-  const ingredientItems = recipeData[0].ingredients.map((ingredient, index) => <li key={index}>{ingredient}</li>);
+export default function RecipeDetail({recipes}) {
+  const params = useParams();
+  const recipe = recipes.filter(recipe => recipe.id === params.id)[0];
```

Comment:
I would recommend to use `find` instead of `filter` here. Then you don't need the square bracket syntax to get the first element. `find` already returns the object.
```suggestion
  const recipe = recipes.find(recipe => recipe.id === params.id);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find